src/sqlparser/src/parser.rs

```diff
@@ -247,17 +219,27 @@ impl Parser {
             let cursor = " ".repeat(prefix.len() + loc.column as usize - 1);
             ParserError::ParserError(format!(
                 "{}\n{}{}\n{}^",
-                e.inner_msg(),
+                e.inner().to_string().replace('\n', ": "),
                 prefix,
                 sql_line,
                 cursor
             ))
         })?;
-        Ok(ast)
+        Ok(stmts)
+    }
+
+    /// Parse object name from a string.
+    pub fn parse_object_name_str(s: &str) -> Result<ObjectName, ParserError> {
+        let mut tokenizer = Tokenizer::new(s);
+        let tokens = tokenizer.tokenize_with_location()?;
+        let parser = Parser(&tokens);
+        Parser::parse_object_name
+            .parse(parser)
```

Comment:
This does not look that intuitive. 🤣 I guess that's because...
- we make combinator a method instead of a bare function
- the name starts with `parse_`

Reply:
Yes. Eventually we should make these methods bare functions without the `parse_` prefix.